dashboard/src/auth/VueAuth.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2020 RisingStack
+import { Vue, Component } from "vue-property-decorator";
+import createAuth0Client, {
+  PopupLoginOptions,
+  Auth0Client,
+  RedirectLoginOptions,
+  GetIdTokenClaimsOptions,
+  GetTokenSilentlyOptions,
+  GetTokenWithPopupOptions,
+  LogoutOptions
+} from "@auth0/auth0-spa-js";
+import { User } from "./User";
+
+export type Auth0Options = {
+  domain: string;
+  clientId: string;
+  audience?: string;
+  [key: string]: string | undefined;
+};
+
+export type RedirectCallback = (appState: { targetUrl: string }) => void;
+
+@Component({})
+export class VueAuth extends Vue {
+  loading = true;
+  isAuthenticated? = false;
+  user?: User;
+  auth0Client?: Auth0Client;
+  popupOpen = false;
+  error?: Error;
+
+  async getUser() {
+    return new User(await this.auth0Client?.getUser());
+  }
+
+  /** Authenticates the user using a popup window */
+  async loginWithPopup(o: PopupLoginOptions) {
+    this.popupOpen = true;
+
+    try {
+      await this.auth0Client?.loginWithPopup(o);
+    } catch (e) {
+      console.error(e);
+      this.error = e;
+    } finally {
+      this.popupOpen = false;
+    }
+
+    this.user = await this.getUser();
+    this.isAuthenticated = true;
+  }
+
+  /** Authenticates the user using the redirect method */
+  loginWithRedirect(o?: RedirectLoginOptions) {
+    return this.auth0Client?.loginWithRedirect(o);
+  }
+
+  /** Returns all the claims present in the ID token */
+  getIdTokenClaims(o?: GetIdTokenClaimsOptions) {
+    return this.auth0Client?.getIdTokenClaims(o);
+  }
+
+  /** Returns the access token. If the token is invalid or missing, a new one is retrieved */
+  getTokenSilently(o?: GetTokenSilentlyOptions) {
+    return this.auth0Client?.getTokenSilently(o);
+  }
+
+  /** Gets the access token using a popup window */
+  getTokenWithPopup(o?: GetTokenWithPopupOptions) {
+    return this.auth0Client?.getTokenWithPopup(o);
+  }
+
+  /** Logs the user out and removes their session on the authorization server */
+  logout(o?: LogoutOptions) {
+    return this.auth0Client?.logout(o);
+  }
+
+  /** Use this lifecycle method to instantiate the SDK client */
+  async init(
+    onRedirectCallback: RedirectCallback,
+    redirectUri: string,
+    auth0Options: Auth0Options
+  ) {
+    // Create a new instance of the SDK client using members of the given options object
+    this.auth0Client = await createAuth0Client({
+      domain: auth0Options.domain,
+      client_id: auth0Options.clientId, // eslint-disable-line @typescript-eslint/camelcase
+      audience: auth0Options.audience,
+      redirect_uri: redirectUri // eslint-disable-line @typescript-eslint/camelcase
```

Comment:
I think we probably want to use refresh tokens and I think auth0 is already configured for them

https://auth0.com/docs/libraries/auth0-single-page-app-sdk#use-rotating-refresh-tokens

Reply:
Yes, I'll add some auth tests and add refresh token use.